### DISCUSSION TITLE
Additional authenticator "Condition - credentials configured"

### DISCUSSION THREAD

**Discussion by danielFesenmeyer** [keycloak/keycloak#21156] [Ideas]
We had to implement a quite complex authentication flow with 2FA, based on OTP and recovery codes. We needed to support conditions such as "OTP _is_ configured" but "Recovery codes _are not_ configured".

In this context, we tried to use the Keycloak builtin authenticator "Condition - user configured", which checks whether the next authenticators are configured for the user (for example, when the next authenticator is OTP form, it checks whether OTP is configured). That's fine for simple conditions, but did not work out well for conditions such as "OTP _is_ configured" but "Recovery codes _are not_ configured".

That's why we implemented a custom authenticator "Condition - credentials configured", which can be configured as follows:

- **Credential type**: _string_, supports a list of values, pre-filled from the providerIds of the CredentialProvider implementations
- **Negate output**: _boolean_, whether the check should be negated (i.e. credentials are not configured)

Is someone interested in this authenticator?
We would be happy to contribute it.

**Comment by dasniko** [comment]
Hi @danielFesenmeyer 
sounds interesting and useful! Why not providing it as a 3rd party community extension and listing it [here](https://www.keycloak.org/extensions.html)?
